qa/rpc-tests/test_framework/util.py

```diff
@@ -99,11 +108,31 @@ def sync_mempools(rpc_connections, wait=1):
 
 bitcoind_processes = {}
 
-def initialize_datadir(dirname, n):
+def initialize_datadir(dirname, n,bitcoinConfDict=None,wallet=None):
     datadir = os.path.join(dirname, "node"+str(n))
     if not os.path.isdir(datadir):
         os.makedirs(datadir)
+    
+    defaults = {"server":1, "discover":0, "regtest":1,"rpcuser":"rt","rpcpassword":"rt",
+                "port":p2p_port(n),"rpcport":str(rpc_port(n)),"listenonion":0,"maxlimitertxfee":0,"blockprioritysize":2000000}
+    if bitcoinConfDict: defaults.update(bitcoinConfDict)
+
     with open(os.path.join(datadir, "bitcoin.conf"), 'w') as f:
+        for (key,val) in defaults.items():
+          if type(val) is type([]):
+            for v in val:
+              f.write("%s=%s\n" % (str(key), str(v)))
+          else:
+            f.write("%s=%s\n"% (str(key), str(val)))
+
+    if wallet:
+      regtestdir = os.path.join(datadir,"regtest")
+      if not os.path.isdir(regtestdir):
+        os.makedirs(regtestdir)
+      shutil.copyfile(wallet,os.path.join(regtestdir, "wallet.dat"))
+      
+
+    if 0:
```

Comment:
Please no debugging sections like this in framework modules.
It makes reviewing tougher as well.
If you need it, put it under some controllable debug parameters.